### ISSUE TITLE
Implement remaining Prometheus config structs

### ISSUE THREAD

**Issue by cboggs** [Fresh-Tracks/bomb-squad#11]
The current alpha implementation of Bomb Squad can only properly re-marshal static and k8s SD scrape configs. This was done initially because:
* A simple `import github.com/prometheus/prometheus/config` will pull in **all** of the vendor packages needed for the various SD mechanisms. This breaks particularly badly when `dep` attempts to pull in the Kubernetes API packages, as those two are wholly incompatible. 
* Reimplementing all of those config structs by hand was a time sink for my schedule at the time

I'm not entirely sure that full (or even partial) re-implementation is necessary, but I've not quite found a better way yet. Reimplementing is easy, if tedious, but I'm concerned with upstream version changes breaking Bomb Squad. Even with a proper import of all the `config` packages from upstream, we'd likely need a `prom-version` command-line arg anyways, so maybe it's fine to reimplement and simply add some version isolation for sanity's sake. I don't imagine the Prometheus config will be a very volatile bit of the codebase, and SD features are under moratorium for now, so this is probably fine for the moment.

**Comment by pcarranza** [comment]
I wouldn't recommend reimplementing the configuration parsing from Prometheus.

In a non-k8s world, we could simply use a single file for the recording rules that are necessary. But I'm not so sure about relabeling.

One possible thing to do is to simply treat the whole configuration of Prometheus as a map of maps and only pay attention to the sections that are necessary, and simply ignore the rest as it will be serialized back again when saving.

Which sections are the ones that are being modified?

**Comment by pcarranza** [comment]
Nah, let's keep files separated, it makes sense because then it's also easier to remove things (just rm the file)

I just checked because I didn't know and there is a [common library](https://github.com/prometheus/common) where the configuration is located, it doesn't look like so bad of a dependency.

**Comment by pcarranza** [comment]
Scratch that, it's only common structs, which means http 😒

**Comment by pcarranza** [comment]
What if we ask?

Else, copying the structs over to an internal package seems to be the least bad approach.

**Comment by cboggs** [comment]
So I opened PR #21 to address this. Going to open another shortly when the Prometheus config pacakge(s) pulled in properly and this goes away. :-)
